### ISSUE TITLE
is space between pages possible (like Hotel Tonight)?

### ISSUE THREAD

**Issue by faceyspacey** [happypancake/react-native-scrollable-tab-view#213]
it seems the `pagingEnabled` param automatically locks you on each page, allowing for proper swiping between pages, but what if you want gaps between the pages. Hotel Tonight does this so you can see the background containing their logo between swipes. 

Any ideas how we can accomplish this?

**Comment by skv-headless** [comment]
Could you please make a video with behaviour that you want to achieve?

**Comment by faceyspacey** [comment]
Ok, I made one. Here it is:

https://www.youtube.com/watch?v=UPi4PwhJ70E

notice the gaps between the pages. That's the goal.

**Comment by skv-headless** [comment]
I don't know good solution.  It would be difficult to implement especially for ios.
Bad solution that might work:
1) make paddings for elements
2) Hide paddings when tab selected

**Comment by irfaan** [comment]
Something like [this](http://stackoverflow.com/a/34094936/1183749)?

**Comment by faceyspacey** [comment]
thanks for this. it's not quite correct because when a page snaps you shouldn't see any padding gaps. Only in between scrolls/swipes should you see it. However, after fiddling it, I 90% got it!:

https://rnplay.org/apps/5Te3bw

I didn't notice the `snapToAlignment` property before, or at least realize its relevance. But basically what I've done is snapped to the `center` and made the cards full width again, and then added margin between the cards, and then snapped to the interval of the card width + 2*margin. 

The one remaining issue is that the first card partially overlaps 2 boundaries. You'll see when the app first loads. The first page is incorrect, but all pages after are correct. I solve a similar issue for the last card by setting its `marginRight` to 0. 

So basically, we just need to solve this for the first card. Basically, there is a "0 card" which should either not be there, or not be reachable. We could probably use state and `onScroll` callbacks to determine if you're trying to reach it, and snap back to the card with index 1. And then in `componentDidMount` we set it to page with index 1. Hopefully that will feel fluid enough--but there may be some other solutions I'm missing.

**Comment by faceyspacey** [comment]
well, it turns out it's easy to solve--in a plain ScrollView, but requires some changes to this package. 

In a plain ScrollView, it's just setting `snapToAlignment='start'` and `snapToInterval` to the page width + the margin you want, and setting each of your pages to have the corresponding `width` and `marginRight`. 

However, *Scrollable Tab View* requires more customization, particularly since it has a TabBar and underline that makes use of the `containerWidth` measurements used. 

Anyway, it's done and I provided a PR for it if anyone wants to check it out:

https://github.com/brentvatne/react-native-scrollable-tab-view/pull/231

You can close this issue. It's iOS only. For now I'll leave the Android version to anyone else that might want this.

**Comment by ptomasroos** [comment]
Closing as this issue is very old, feel free to open a new issue if its still happening in the latest release
